Modules/_io/fileio.c

```diff
@@ -447,6 +447,31 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
         }
     }
     else {
+#if defined(S_ISREG) && defined(ENOTDIR)
+        /* On AIX and Windows, open() may succeed for files with a trailing slash.
+           The Open Group specifies filenames ending with a trailing slash should
+           be an error - ENOTDIR */
+        if ( (fd < 0) && S_ISREG(fdfstat.st_mode)) {
+
+            int trailing_slash = 0;
```

Comment:
On second thought, I think this code is simple enough to eliminate `trailing_slash`. Also, per PEP 7, the line length should be less than 80 characters, and there should be no space after the left paren (e.g. `if ((fd < 0)` or just `if (fd < 0`, not `if ( (fd < 0)`). 

```C
        if (fd < 0 && S_ISREG(fdfstat.st_mode)) {
#ifdef MS_WINDOWS
            size_t len = wcslen(widename);
            if (len && (widename[len - 1] == L'\\' ||
                widename[len - 1] == L'/')) {
#else
            size_t len = strlen(name);
            if (len && name[len - 1] == '/') {
#endif
                errno = ENOTDIR;
                PyErr_SetFromErrnoWithFilenameObject(PyExc_OSError, nameobj);
                goto error;
            }
        }
#endif
```

Reply:
trailing_slash and other points modified as requested.